File: iosApp/iosApp/DesignSystem/Components/TrenifyStatusPill.swift
import SwiftUI

/// Visual treatment, not domain state. Native consumers map shared semantic outputs and supply labels.
enum TrenifyStatusTone: CaseIterable {
    case onTime, delayed, cancelled, arrived, warning, unknown

    var symbol: String {
        switch self {
        case .onTime: return "checkmark.circle.fill"
        case .delayed: return "clock.badge.exclamationmark"
        case .cancelled: return "xmark.octagon.fill"
        case .arrived: return "flag.checkered"
        case .warning: return "exclamationmark.triangle.fill"
        case .unknown: return "questionmark.circle"
        }
    }

    var color: Color {
        switch self {
        case .onTime: return TrenifyColors.statusOnTime
        case .delayed: return TrenifyColors.statusDelayed
        case .cancelled: return TrenifyColors.statusCancelled
        case .arrived: return TrenifyColors.statusArrived
        case .warning: return TrenifyColors.statusWarning
        case .unknown: return TrenifyColors.textSecondary
        }
    }
}

struct TrenifyStatusPill: View {
    var tone: TrenifyStatusTone
    var label: String
    var accessibilityLabel: String? = nil

    var body: some View {
        Label {
            Text(label).fixedSize(horizontal: false, vertical: true)
        } icon: {
            Image(systemName: tone.symbol).accessibilityHidden(true)
        }
        .font(TrenifyTypography.status)
        .foregroundStyle(tone.color)
        .padding(.horizontal, TrenifySpacing.m)
        .padding(.vertical, TrenifySpacing.s)
        .background(TrenifyColors.surfaceMuted, in: TrenifyShapes.statusPill)
        .accessibilityElement(children: .ignore)
        .accessibilityLabel(Text(accessibilityLabel ?? label))
    }
}

File: androidApp/src/main/kotlin/it/danielebufarini/trenify/design/component/TrenifyStatusPill.kt
package it.danielebufarini.trenify.design.component

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.semantics.clearAndSetSemantics
import androidx.compose.ui.semantics.contentDescription
import it.danielebufarini.trenify.design.theme.TrenifyColors
import it.danielebufarini.trenify.design.theme.TrenifyShapes
import it.danielebufarini.trenify.design.theme.TrenifySpacing
import it.danielebufarini.trenify.design.theme.TrenifyTheme

/** Visual treatment only. Consumers supply the authoritative semantic label; no delay/status inference. */
enum class TrenifyStatusTone(val symbol: String) {
    OnTime("✓"), Delayed("+"), Cancelled("×"), Arrived("⚑"), Warning("!"), Unknown("?"),
}

fun TrenifyStatusTone.color(colors: TrenifyColors): Color = when (this) {
    TrenifyStatusTone.OnTime -> colors.statusOnTime
    TrenifyStatusTone.Delayed -> colors.statusDelayed
    TrenifyStatusTone.Cancelled -> colors.statusCancelled
    TrenifyStatusTone.Arrived -> colors.statusArrived
    TrenifyStatusTone.Warning -> colors.statusWarning
    TrenifyStatusTone.Unknown -> colors.textSecondary
}

@Composable
fun TrenifyStatusPill(
    tone: TrenifyStatusTone,
    label: String,
    modifier: Modifier = Modifier,
    accessibilityLabel: String = label,
) {
    Surface(
        modifier = modifier.clearAndSetSemantics { contentDescription = accessibilityLabel },
        shape = TrenifyShapes.statusPill, color = TrenifyTheme.colors.surfaceMuted,
        contentColor = tone.color(TrenifyTheme.colors),
    ) {
        Row(
            Modifier.padding(horizontal = TrenifySpacing.m, vertical = TrenifySpacing.s),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(TrenifySpacing.s),
        ) {
            Text(tone.symbol, style = TrenifyTheme.typography.status)
            Text(label, style = TrenifyTheme.typography.status)
        }
    }
}
